From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yingkai Song <Y.Song-21@student.tudelft.nl>
Date: Thu, 24 Mar 2022 22:17:29 +0100
Subject: [PATCH] 4100: replace error-prone '==' with 'equals()'

Fix the possibility of NullPointerException by calling .equals() using constant instead of a variable
---
 .../vt/gui/provider/VTFunctionReferenceCorrelator_ELF_Test.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Ghidra/Features/VersionTracking/src/test.slow/java/ghidra/feature/vt/gui/provider/VTFunctionReferenceCorrelator_ELF_Test.java b/Ghidra/Features/VersionTracking/src/test.slow/java/ghidra/feature/vt/gui/provider/VTFunctionReferenceCorrelator_ELF_Test.java
index 25dccabb4..6c50539fa 100644
--- a/Ghidra/Features/VersionTracking/src/test.slow/java/ghidra/feature/vt/gui/provider/VTFunctionReferenceCorrelator_ELF_Test.java
+++ b/Ghidra/Features/VersionTracking/src/test.slow/java/ghidra/feature/vt/gui/provider/VTFunctionReferenceCorrelator_ELF_Test.java
@@ -105,7 +105,7 @@ public class VTFunctionReferenceCorrelator_ELF_Test extends AbstractVTCorrelator
 		List<VTMatchSet> exactMatchSets = session.getMatchSets();
 		for (VTMatchSet ms : exactMatchSets) {
 			String corrName = ms.getProgramCorrelatorInfo().getName();
-			if (corrName == exactSymbolNameCorrelator) {
+			if (exactSymbolNameCorrelator.equals(corrName)) {
 				ApplyMatchTask task =
 					new ApplyMatchTask(controller, (List<VTMatch>) ms.getMatches());
 				runTask(task);
-- 
2.43.0

